vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -706,10 +713,17 @@ type internal FSharpNavigation(metadataAsSource: FSharpMetadataAsSourceService,
             let gtd = GoToDefinition(metadataAsSource)
             let! result = gtd.FindDefinitionAtPosition(initialDoc, position)
 
-            return
-                match result with
-                | ValueSome(FSharpGoToDefinitionResult.NavigableItem(navItem), _) -> ImmutableArray.create navItem
-                | _ -> ImmutableArray.empty
+            match result with
+            | ValueSome(FSharpGoToDefinitionResult.NavigableItem(navItem), _) -> return ImmutableArray.create navItem
+            | ValueSome(FSharpGoToDefinitionResult.ExternalAssembly(targetSymbolUse, metadataReferences), _) ->
+                let! res = gtd.NavigateToExternalDeclarationAsync(targetSymbolUse, metadataReferences)
+
+                if not res then
```

Comment:
I'd rather add a comment before `NavigateToExternalDeclarationAsync` (on why it is needed) and then just have it like:
```fsharp
// force external navigation and hope for the best
let! _ = gtd.NavigateToExternalDeclarationAsync(targetSymbolUse, metadataReferences)
return ImmutableArray.empty
```

Sending results to telemetry or playing with cancelables can be an option, but minimal fix to keep things working also seems appropriate to me.